container/view.go

```diff
@@ -523,22 +492,21 @@ type namesByContainerIDIndexer struct{}
 func (e *namesByContainerIDIndexer) FromObject(obj interface{}) (bool, []byte, error) {
 	n, ok := obj.(nameAssociation)
 	if !ok {
-		return false, nil, fmt.Errorf(`%T does not have type "nameAssociation"`, obj)
+		return false, nil, errdefs.InvalidParameter(errors.Errorf(`%T does not have type "nameAssociation"`, obj))
```

Comment:
Is this error something that the user can do something about or would such an error only be produced by a bug in the caller?

Reply:
I removed the last commit; need to look more in-depth;

While these errors are "invalid parameter", the cause of the invalid parameter (wrong type) is not due to what the user did.

I'll revisit this one, and have a closer look;

- Either: keep InvalidParameter, but make sure it's wrapped to "system" after that
- Or (this commit); consider it "system" errors from the start


It's slightly difficult to follow where these are called, as it's implementing an interface, and that interface is called from the vendored memdb code, so need to look closer